x-pack/plugins/lens/public/indexpattern_datasource/datapanel.tsx

```diff
@@ -488,23 +561,83 @@ export const InnerIndexPatternDataPanel = function InnerIndexPatternDataPanel({
         responsive={false}
       >
         <EuiFlexItem grow={false}>
-          <div className="lnsInnerIndexPatternDataPanel__header">
-            <ChangeIndexPattern
-              data-test-subj="indexPattern-switcher"
-              trigger={{
-                label: currentIndexPattern.title,
-                title: currentIndexPattern.title,
-                'data-test-subj': 'indexPattern-switch-link',
-                fontWeight: 'bold',
-              }}
-              indexPatternId={currentIndexPatternId}
-              indexPatternRefs={indexPatternRefs}
-              onChangeIndexPattern={(newId: string) => {
-                onChangeIndexPattern(newId);
-                clearLocalState();
-              }}
-            />
-          </div>
+          <EuiFlexGroup
+            gutterSize="s"
+            alignItems="center"
+            className="lnsInnerIndexPatternDataPanel__header"
+          >
+            <EuiFlexItem grow={true}>
+              <ChangeIndexPattern
+                data-test-subj="indexPattern-switcher"
+                trigger={{
+                  label: currentIndexPattern.title,
+                  title: currentIndexPattern.title,
+                  'data-test-subj': 'indexPattern-switch-link',
+                  fontWeight: 'bold',
+                }}
+                indexPatternId={currentIndexPatternId}
+                indexPatternRefs={indexPatternRefs}
+                onChangeIndexPattern={(newId: string) => {
+                  onChangeIndexPattern(newId);
+                  clearLocalState();
+                }}
+              />
+            </EuiFlexItem>
+            {addField && (
+              <EuiFlexItem grow={false}>
+                <EuiPopover
+                  panelPaddingSize="s"
+                  isOpen={popoverOpen}
+                  closePopover={() => {
+                    setPopoverOpen(false);
+                  }}
+                  data-test-subj="lnsIndexPatternActions-popover"
+                  button={
+                    <EuiButtonIcon
+                      iconType="boxesVertical"
+                      data-test-subj="lnsIndexPatternActions"
+                      onClick={() => {
+                        setPopoverOpen(!popoverOpen);
+                      }}
+                    />
```

Comment:
Small icon and color tweak to more closely match the design.

```suggestion
                    <EuiButtonIcon
                      color="text"
                      iconType="boxesHorizontal"
                      data-test-subj="lnsIndexPatternActions"
                      onClick={() => {
                        setPopoverOpen(!popoverOpen);
                      }}
                    />
```